### PR TITLE
113 transaction hash during signin

### PR DESCRIPTION
**Pull request by J4ckyLIM**
- Fix typo
- Change modal message during signMessage
- Add toastpromise during connect function

### CODE REVIEW

File: src/App/Routes/Modals/RegisterModal/GithubSignin/index.tsx [hunk 1]
```diff
@@ -47,16 +50,21 @@ const GithubSigninContainer: FC<PropsWithChildren<Props>> = ({ children, classNa
         signature,
       });
 
+      toast.success("Github account successfully validated");
+
       setIsRegistering(false);
+      setTxHash(hash);
```
Comment (ltoussaint) [inline src/App/Routes/Modals/RegisterModal/GithubSignin/index.tsx:56]: The `hash` variable is not a transaction hash but a simple hash of the message (which is a string) we want to sign. It's used by our backend to verify the signature.

File: src/App/Routes/Modals/RegisterModal/GithubSignin/View.tsx [hunk 2]
```diff
@@ -46,17 +48,23 @@ const GithubSignin: FC<PropsWithChildren<Props>> = ({
...
   );
 
-  function renderLoadingModal() {
+  function renderLoadingModal(txHash?: string) {
     return (
       <Modal contentClassName="px-16 pt-6 pb-8 z-100" isOpen={isLoading} onClose={onClose}>
         <div className="flex flex-col justify-center items-center text-4xl text-center text-blue-500 mt-12">
           <Loader className="animate-spin mr-4 mb-4" size={62} />
           <div>We are verifying your information</div>
+          {txHash && (
```
Comment (ltoussaint) [inline src/App/Routes/Modals/RegisterModal/GithubSignin/View.tsx:62]: You don't need this as there is no transaction ongiong.
Just change the message and close the modal when the user signed (or reject) the message

**Review by ltoussaint — CHANGES_REQUESTED** [review]
Hello @J4ckyLIM 

The task definition is not up to date.
The process does not involve a transaction anymore, so there is no need to display a transaction hash which does not exist.

What is needed :
- display a good message while the user is signing the message (which is part of the verification process that the account is yours). So when the signin window from ArgentX or Braavos appears, we should display a modal like "We need you to sign a message using your wallet to ensure your are the real owner. Then we will be able to link your Github account with your Starknet account."
- The second step include the call to our api to perform the connection between those 2 accounts. At this point you need to use the `toastPromise` method to manage ongoing, success and error messages.

**Comment by J4ckyLIM** [comment]
> Hello @J4ckyLIM
> 
> The task definition is not up to date. The process does not involve a transaction anymore, so there is no need to display a transaction hash which does not exist.
> 
> What is needed :
> 
> * display a good message while the user is signing the message (which is part of the verification process that the account is yours). So when the signin window from ArgentX or Braavos appears, we should display a modal like "We need you to sign a message using your wallet to ensure your are the real owner. Then we will be able to link your Github account with your Starknet account."
> * The second step include the call to our api to perform the connection between those 2 accounts. At this point you need to use the `toastPromise` method to manage ongoing, success and error messages.

- I see, but we're currently displaying a modal that displays the status of the of the process, does it mean we show an another modal before displaying the "process" modal ? 
- What call should be done to perform the connection between the 2 accounts ?

**Comment by ltoussaint** [comment]
> * I see, but we're currently displaying a modal that displays the status of the of the process, does it mean we show an another modal before displaying the "process" modal ?
> * What call should be done to perform the connection between the 2 accounts ?

- No, the current modal must stay, just change the message. The only difference is that it should be closed earlier than today (when the user signed the message, we close the modal)
- The call to connect the account already exists, it is on `src/app/routes/modals/RegisterModal/GithubSignin/index.tsx#43` with the function `connect`.

File: src/App/Routes/Modals/RegisterModal/GithubSignin/View.tsx [hunk 3]
```diff
@@ -60,7 +60,11 @@ const GithubSignin: FC<PropsWithChildren<Props>> = ({
       <Modal contentClassName="px-16 pt-6 pb-8 z-100" isOpen={isLoading} onClose={onClose}>
         <div className="flex flex-col justify-center items-center text-4xl text-center text-blue-500 mt-12">
           <Loader className="animate-spin mr-4 mb-4" size={62} />
-          <div>We are verifying your information</div>
+          <div>
+            We need you to sign a message using your wallet to ensure your are the real owner. Then we will be able to
+            link your Github account with your Starknet account.
+          </div>
+          <div>Estimated remaining time: 5 mins</div>
```
[the remark this answers is not in the document]
Comment (J4ckyLIM) [reply src/App/Routes/Modals/RegisterModal/GithubSignin/View.tsx:67]: Oh yeah true

**Review by ltoussaint — APPROVED** [review]

**Comment by ltoussaint** [comment]
@J4ckyLIM you have to rebase the `develop` branch into yours
Make sure to rebase from the upstream branch (the one from OnlyDust repo) as we had to perform a force push on develop few days ago

**Pull request closed without merging**